### ISSUE TITLE
Make name generation unique

### ISSUE THREAD

**Issue by holmgr** [holmgr/gemini#11]
### Description

Currently the implementation of a name generator using the MarkovChain can and will not only produce string which may be in the training set, but also string which have previously been generated.
As unique names are of high importance, and also that the training data maybe names from other sci-fi universes, the generator cannot be allowed to generate them.

A suggestion would be to implement a type of iterator producing all possible names from the training set, subtracting the training set and then returning a shuffled iterator which can be used to get an number of names until no unique name can be generated anymore.
At that point scientific names probably should be the fallback.

### Steps to Reproduce
Produce a few number of strings from MarkovChain using a limited training set.

**Expected behavior:** [What you expect to happen]
Only unique strings, not available in the training data

**Actual behavior:** [What actually happens]
Non unique string, more often than note available in the training set

**Reproduces how often:** [What percentage of the time does it reproduce?]
Depends on the training set

### Versions
N/A

### Additional Information
N/A
